kernel/src/main/java/org/kframework/parser/json/JsonParser.java

```diff
@@ -0,0 +1,103 @@
+import java.io.UnsupportedEncodingException;
+import java.io.StringReader;
+import java.util.Arrays;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+/**
+ * Parses a Json term into the KORE data structures.
+ */
+public class JsonParser {
+
+    public static final String KTOKEN         = "KToken"
+                             , KAPPLY         = "KApply"
+                             , KSEQUENCE      = "KSequence"
+                             , KVARIABLE      = "KVariable"
+                             , KREWRITE       = "KRewrite"
+                             , INJECTEDKLABEL = "InjectedKLabel"
+                             ;
+
+    public static K parse(byte[] data) throws UnsupportedEncodingException {
+        return parse(new String(data, "UTF-8"));
+    }
+
+    public static K parse(String data) {
+        JsonReader reader = Json.createReader(new StringReader(data));
+        return parseJson(reader.readObject());
+    }
+
+    public static K parseJson(JsonObject data) {
+        try {
+            if (! (data.containsKey("format") && data.containsKey("version") && data.containsKey("term"))) {
+                throw KEMException.criticalError("Must have `format`, `version`, and `term` fields in serialized Json!");
+            }
+            if (! data.getString("format").equals("KAST")) {
+                throw KEMException.criticalError("Only can deserialize 'KAST' format Json! Found: " + data.getString("format"));
+            }
+            if (data.getInt("version") != 1) {
+                throw KEMException.criticalError("Only can deserialize KAST version '1'! Found: " + data.getInt("version"));
+            }
+            return toK(data.getJsonObject("term"));
+        } catch (IOException e) {
+            throw KEMException.criticalError("Could not read K term from json", e);
+        }
+    }
+
+    private static K toK(JsonObject data) throws IOException {
+        switch (data.getString("node")) {
+
+            case KTOKEN:
+                return new KToken(data.getString("token"), Outer.parseSort(data.getString("sort")));
+
+            case KAPPLY:
+                int arity = data.getInt("arity");
+                K[] args  = toKs(arity, data.getJsonArray("args"));
+                return KApply.of(KLabel(data.getString("label")), args);
+
+            case KSEQUENCE:
+                int seqLen = data.getInt("arity");
+                K[] items  = toKs(seqLen, data.getJsonArray("items"));
+                return new KSequence(items);
+
+            case KVARIABLE:
+                return new KVariable(data.getString("name"));
+
+            case KREWRITE:
+                K lhs = toK(data.getJsonObject("lhs"));
+                K rhs = toK(data.getJsonObject("rhs"));
+                return new KRewrite(lhs, rhs);
+
+            case INJECTEDKLABEL:
+                return new InjectedKLabel(KLabel(data.getString("name")));
+
```

Comment:
If you want to be able to serialize and deserialize arbitrary rules, you need to also include a case for KAs.